otel-agent/ecs-ec2/ecsattributesprocessor/config.go

```diff
@@ -0,0 +1,58 @@
+package ecsattributesprocessor
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+)
+
+type Config struct {
+	// Attributes is a list of attribute pattern to be added to the resource
+	Attributes []string `mapstructure:"attributes"`
+
+	// Source fields specifies what resource attribute field to read the
+	// container ID
+	// container_id:
+	//   sources:
+	//     - "log.file.name"
+	// 	   - "container.id"
+	ContainerID     `mapstructure:"container_id"`
+	attrExpressions []*regexp.Regexp // store compiled regexes
+}
+
+type ContainerID struct {
+	Sources []string `mapstructure:"sources"`
+}
+
+func (c *Config) validate() error {
+	// check ContainerID sources
+	if len(c.ContainerID.Sources) == 0 {
+		return errors.New("atleast one container ID source must be specified. [container_id.sources]")
+	}
+
+	// validate attribute regex
+	for _, expr := range c.Attributes {
+		r, err := regexp.Compile(expr)
+		if err != nil {
+			return fmt.Errorf("invalid expression found under attributes pattern %s - %s", expr, err)
+		}
+
+		c.attrExpressions = append(c.attrExpressions, r)
```

Comment:
Updated this function name to `init()` since it serves a dual purpose. If we remove the functionality and do it elsewhere, we end up iterating and compiling the same regex slice twice. Yes, the overhead there is small, but since we're already doing it here, we may as well store the response.